### PR TITLE
Fix spritepolygon

### PR DESCRIPTION
**Pull request by Wu-Hao**
PolygonSprite is now removed, 
Polygon functionality merged into CCSprite, 
PolygonCache removed,
Added AutoPolygon helper class

### CODE REVIEW

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by fusijie** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
@Wu-Hao  please solve merge conflicts. thanks.

**Comment by pandamicro** [comment]
I will commit necessary JSB modifications today

**Comment by fusijie** [comment]
[ci rebuild]

**Comment by fusijie** [comment]
[ci rebuild]

**Comment by pandamicro** [comment]
Still a link issue on linux: 

```
../../lib/libcocos2d.a(CCAutoPolygon.cpp.o): In function `ClipperLib::PolyTree::PolyTree()':
/linux_slave/workspace/ccs-pr/node/linux/external/clipper/clipper.hpp:155: undefined reference to `ClipperLib::PolyNode::PolyNode()'
collect2: error: ld returned 1 exit status
tests/cpp-tests/CMakeFiles/cpp-tests.dir/build.make:4782: recipe for target 'bin/cpp-tests/cpp-tests' failed
```

It can be merged once linux passed the compilation

File: cocos/2d/CCAutoPolygon.cpp [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+THE SOFTWARE.
+****************************************************************************/
+
+#include "CCAutoPolygon.h"
+#include "poly2tri/poly2tri.h"
+#include "base/CCDirector.h"
+#include "renderer/CCTextureCache.h"
+#include "clipper/clipper.hpp"
+#include <algorithm>
+#include <math.h>
```
Comment (zilongshanren) [inline cocos/2d/CCAutoPolygon.cpp:34]: The include order should be:  c header files, c++ header file, 3rd party library files and then project header file.

**Pull request merged**